commit 8845b3243fcd
Author: Lee Salzman <lsalzman@mozilla.com>
Date:   Tue Oct 10 15:49:13 2017 -0400

    Bug 1387079 - Only use SkRasterPipeline when SkJumper is accelerated. r=jrmuizel, a=ritu
    
    MozReview-Commit-ID: 13dd2Ap4sob
    
    --HG--
    extra : source : 32227656b09dfb5d7c5412c799c7081dbc49346c
---
 gfx/skia/skia/src/core/SkBlitter.cpp                                  | 2 ++
 .../mozilla/mozilla-central-reftests/masking/mask-mode-d.html.ini     | 4 +++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git gfx/skia/skia/src/core/SkBlitter.cpp gfx/skia/skia/src/core/SkBlitter.cpp
index 0e8bd2db2249..3a824bb724f7 100644
--- gfx/skia/skia/src/core/SkBlitter.cpp
+++ gfx/skia/skia/src/core/SkBlitter.cpp
@@ -847,9 +847,11 @@ SkBlitter* SkBlitter::Choose(const SkPixmap& device,
         return alloc->make<SkA8_Coverage_Blitter>(device, *paint);
     }
 
+#if (!defined(SK_JUMPER_USE_ASSEMBLY) || SK_JUMPER_USE_ASSEMBLY) && (defined(__aarch64__) || defined(__arm__) || defined(__x86_64__) || defined(_M_X64))
     if (SkBlitter* blitter = SkCreateRasterPipelineBlitter(device, *paint, matrix, alloc)) {
         return blitter;
     }
+#endif
 
     if (nullptr == shader) {
         if (mode != SkBlendMode::kSrcOver) {
diff --git testing/web-platform/meta/css/vendor-imports/mozilla/mozilla-central-reftests/masking/mask-mode-d.html.ini testing/web-platform/meta/css/vendor-imports/mozilla/mozilla-central-reftests/masking/mask-mode-d.html.ini
index c1246ddb4127..a02e049c3684 100644
--- testing/web-platform/meta/css/vendor-imports/mozilla/mozilla-central-reftests/masking/mask-mode-d.html.ini
+++ testing/web-platform/meta/css/vendor-imports/mozilla/mozilla-central-reftests/masking/mask-mode-d.html.ini
@@ -1,3 +1,5 @@
 [mask-mode-d.html]
   type: reftest
-  expected: FAIL
+  expected:
+    if (processor == "x86") and (bits == 32): PASS
+    FAIL
